runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -11797,6 +11797,357 @@ static TR::Register *inlineEncodeUTF16(TR::Node *node, TR::CodeGenerator *cg)
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi, node, result, result, -16);
+
+   // Finally, combine this with the address of the last vector load to find the address of the
+   // first matching element in the string. Finally, use this to calculate the corresponding index.
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::add, node, result, result, currentAddress);
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::subf, node, result, arrAddress, result);
+
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, foundExactLabel);
+   if (!isLatin1)
+      generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::srawi, node, result, result, 1);
+
+      {
+      TR::RegisterDependencyConditions *deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 15 + srm->numAvailableRegisters(), cg->trMemory());
+
+      deps->addPostCondition(arrObjectAddress, TR::RealRegister::NoReg);
+      deps->getPostConditions()->getRegisterDependency(0)->setExcludeGPR0();
+      deps->addPostCondition(targetScalar, TR::RealRegister::NoReg);
+      if (node->getChild(2)->getReferenceCount() == 1)
+         deps->getPostConditions()->getRegisterDependency(1)->setExcludeGPR0();
+      deps->addPostCondition(startIndex, TR::RealRegister::NoReg);
+      if (node->getChild(3)->getReferenceCount() == 1)
+         deps->getPostConditions()->getRegisterDependency(2)->setExcludeGPR0();
+      deps->addPostCondition(endIndex, TR::RealRegister::NoReg);
+      if (node->getChild(4)->getReferenceCount() == 1)
+         deps->getPostConditions()->getRegisterDependency(3)->setExcludeGPR0();
+
+      deps->addPostCondition(cr0, TR::RealRegister::cr0);
+      deps->addPostCondition(cr6, TR::RealRegister::cr6);
+
+      deps->addPostCondition(zeroRegister, TR::RealRegister::NoReg);
+      deps->addPostCondition(result, TR::RealRegister::NoReg);
+      deps->getPostConditions()->getRegisterDependency(7)->setExcludeGPR0();
+      deps->addPostCondition(arrAddress, TR::RealRegister::NoReg);
+      deps->addPostCondition(currentAddress, TR::RealRegister::NoReg);
+      deps->getPostConditions()->getRegisterDependency(9)->setExcludeGPR0();
+      deps->addPostCondition(endAddress, TR::RealRegister::NoReg);
+
+      deps->addPostCondition(targetVector, TR::RealRegister::NoReg);
+      deps->addPostCondition(targetVectorNot, TR::RealRegister::NoReg);
+      deps->addPostCondition(searchVector, TR::RealRegister::NoReg);
+      deps->addPostCondition(permuteVector, TR::RealRegister::NoReg);
+
+      srm->addScratchRegistersToDependencyList(deps, true);
+
+      generateDepLabelInstruction(cg, TR::InstOpCode::label, node, endLabel, deps);
+
+      deps->stopUsingDepRegs(cg, result);
+
+      cg->decReferenceCount(node->getChild(0));
+      cg->decReferenceCount(node->getChild(1));
+      cg->decReferenceCount(node->getChild(2));
+      cg->decReferenceCount(node->getChild(3));
+      cg->decReferenceCount(node->getChild(4));
+      }
+
+   node->setRegister(result);
```

Comment:
FYI though it can't happen here because you allocated the register it's usually good practice to set the node's register _before_ calling `decReferenceCount` to avoid the situation where the `result` is obtained by calling `evaluate(...)` on one of `node`'s children. In such a scenario if you call `decReferenceCount` before `setRegister` you could end up killing a register because it's use count decrements to 0.